### DISCUSSION TITLE
Toggling transparency (for a tree) creates a black background for several seconds

### DISCUSSION THREAD

**Discussion by alexlitinsky** [LazyVim/LazyVim#203] [General]
I'm using these commands in the init.lua file to enable transparency:
vim.cmd("hi NonText guibg=NONE ctermbg=NONE")
vim.cmd("hi Normal guibg=NONE ctermbg=NONE")
vim.cmd("hi NeoTreeNormal guibg=NONE ctermbg=NONE")
vim.cmd("hi NeoTreeNormalNC guibg=NONE ctermbg=NONE")
vim.cmd("hi NormalNC guibg=none ctermbg=none") -- credit to folke 
vim.cmd("hi LineNr guibg=none ctermbg=none")
vim.cmd("hi Folded guibg=none ctermbg=none")
vim.cmd("hi SpecialKey guibg=none ctermbg=none")
vim.cmd("hi VertSplit guibg=none ctermbg=none")
vim.cmd("hi SignColumn guibg=none ctermbg=none")
vim.cmd("hi EndOfBuffer guibg=none ctermbg=none")

However, if I toggle a new buffer, like a tree or a split window, the background of that buffer becomes black for half a second. Is there a way to prevent this?

**Comment by folke** [comment]
Probably because you need to set transparent to another highlight group. no idea

**Comment by loiccoyle** [comment]
Maybe you could try to use a plugin like, [xiyaowong/nvim-transparent](https://github.com/xiyaowong/nvim-transparent) it might solve your issue.

**Comment by folke** [comment]
Most colorschemes also have options to make them transparent. For TokyoNight:

```lua
  {
    "folke/tokyonight.nvim",
    opts = {
      transparent = true,
      styles = {
        sidebars = "transparent",
        floats = "transparent",
      },
    },
  },
```

**Reply by alexlitinsky** [reply]
Where specifically would we add this code in our lazyvim? Would we add this as a plugin or as an autocmd?

**Reply by folke** [reply]
you add it under `lua/plugins/whatever.lua`

You should realy read the docs, since that is a really basic thing I posted.

The docs are here https://www.lazyvim.org/

**Reply by alexlitinsky** [reply]
The reason I wanted clarification on this is that when I did return { code } in a new file in the plugins directory, lazy should see that the plugin isn't installed and I can install but that option isn't popping up.

**Reply by folke** [reply]
tokyonight is already part of LazyVim, so already installed. Not really sure what you mean.

Again, please read the docs. I made them for a reason.

**Reply by alexlitinsky** [reply]
Oh, I see. The plugin worked in that everything was transparent but I still encountered the error where the background of the buffer that I just left is black for several seconds. I also tried the plugin mentioned above and that didn't work as well.

**Comment by alexlitinsky** [comment]
Here's a better visualization of what I'm talking about. 

https://user-images.githubusercontent.com/86020820/217318418-4b00b830-2ff3-4a98-8fa0-1aecded0255b.mov

**Reply by folke** [reply]
I see now. It's caused by `mini.animate`. If you disable it, the problem will go away.

Best to check with the mini.animate repo to see how you can solve that. Make sure to add the video to make it clear.

**Reply by alexlitinsky** [reply]
Will do. Thanks for the help!

**Reply by felipe300** [reply]
If you disable the plugin using
`return {
    { "echasnovski/mini.animate", enabled = true}
}`, the error goes away, but i couldn't solve. did you solve it @alexlitinsky ?

**Reply by alexlitinsky** [reply]
Yes, it worked.

**Reply by folke** [reply]
@felipe300 you need to set `enabled=false`, to disable it
